File: CaveEscape/src/CaveEscape3.pas
program GameMain;
uses SwinGame, sgTypes, sgTimers, sgSprites, sysUtils;

const
	GRAVITY = 0.08;
	MAX_SPEED = 5;

function GetNewPlayer(): Sprite;
begin
	result := CreateSprite(BitmapNamed('Player'), AnimationScriptNamed('PlayerAnimations'));
	SpriteSetX(result, ScreenWidth() / 2 - SpriteWidth(result));
	SpriteSetY(result, ScreenHeight() / 2);
	SpriteStartAnimation(result, 'Fly');
end;

procedure UpdateVelocity(var toUpdate: Sprite);
begin
	SpriteSetDy(toUpdate, SpriteDy(toUpdate) + GRAVITY);
	if SpriteDy(toUpdate) > MAX_SPEED then
	begin
		SpriteSetDy(toUpdate, MAX_SPEED);
	end
	else if (SpriteDy(toUpdate) < -(MAX_SPEED)) then
	begin
		 SpriteSetDy(toUpdate,	-(MAX_SPEED));
	end;
end;

procedure Main();
var
	player: Sprite;
begin
	OpenGraphicsWindow('Cave Escape', 432, 768);
	OpenAudio();
	LoadResourceBundleNamed('CaveEscape', 'CaveEscape.txt', false);

	player := GetNewPlayer();

	repeat // The game loop...
		ProcessEvents();
		ClearScreen(ColorWhite);

		UpdateVelocity(player);
		UpdateSprite(player);

		DrawSprite(player);

		RefreshScreen();

	until WindowCloseRequested();
end;

begin
	Main();
end.

File: C/CaveEscape/src/cave_escape_3.cpp
#include <stdio.h>
#include "SwinGame.h"

#define GRAVITY 0.08
#define MAX_SPEED 5

sprite get_new_player()
{
  sprite result;
  result = create_sprite(bitmap_named("Player"), animation_script_named("PlayerAnimations"));
  sprite_set_x(result, screen_width() / 2 - sprite_width(result));
  sprite_set_y(result, screen_height() / 2);
  sprite_start_animation(result, "Fly");

  return result;
}

void update_velocity(sprite player)
{
  sprite_set_dy(player, sprite_dy(player) + GRAVITY);

  if (sprite_dy(player) > MAX_SPEED)
  {
    sprite_set_dy(player, MAX_SPEED);
  }
  else if (sprite_dy(player) < -(MAX_SPEED))
  {
    sprite_set_dy(player, -(MAX_SPEED));
  }
}

int main()
{
    sprite player;

    open_graphics_window("Cave Escape", 432, 768);
    load_resource_bundle_named("CaveEscape", "CaveEscape.txt", false);

    player = get_new_player();

    do // The game loop...
    {
      process_events();
      clear_screen(COLOR_WHITE);

      update_velocity(player);
      update_sprite(player);
      draw_sprite(player);

      refresh_screen();

    } while(!window_close_requested());

    return 0;
}
